scintillometry/phases/phase.py

```diff
@@ -0,0 +1,667 @@
+# -*- coding: utf-8 -*-
+# Licensed under the GPLv3 - see LICENSE
+"""Provide a Phase class with integer and fractional part."""
+import operator
+
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import Angle, Longitude
+from astropy.time import Time
+from astropy.time.utils import two_sum, two_product
+from astropy.utils import minversion
+
+
+__all__ = ['Phase', 'FractionalPhase']
+
+
+NUMPY_LT_1_16 = not minversion('numpy', '1.16')
+
+FRACTION_UFUNCS = {np.cos, np.sin, np.tan, np.spacing}
+
+COMPARISON_UFUNCS = {
+    np.equal, np.not_equal,
+    np.less, np.less_equal, np.greater, np.greater_equal}
+
+
```

Comment:
Do you mean a string `fffffff.fffffffffffffffff'? Ie., decoding it like for [DADA times](https://github.com/mhvk/baseband/blob/master/baseband/dada/header.py#L377)?

It would not be difficult to do. Would a classmethod be OK, or do you think the initializer itself should deal with it? (Slightly prefer the former myself).

Reply:
Yes, decoding it like DADA times, or a big number of phase. From the point of the user,  I think the initializer one would be handy.

Reply:
I like the idea, but think we should then also allow writing such long strings. So, I made it a separate issue: gh-93